### PR TITLE
Update CI to run different workloads for commit/PR/release

### PR DESCRIPTION
**Pull request by morenod**
Commit example: https://github.com/morenod/kube-burner/actions/runs/5156181231
PR example: https://github.com/morenod/kube-burner/pull/7
PR workload: https://github.com/morenod/kube-burner/actions/runs/5156204266
Release example: https://github.com/morenod/kube-burner/releases/tag/4.2.0
Release workload: https://github.com/morenod/kube-burner/actions/runs/5156556435

Docs Example: https://morenod.github.io/kube-burner/4.2.0/
Quay Images: https://quay.io/repository/morenod/kube-burner?tab=tags

### CODE REVIEW

File: test/test-ocp.bats [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+@test "Running node-density-cni wrapper with gc=false" {
+  # Disable gc and avoid metric indexing
+  run kube-burner ocp node-density-cni --pods-per-node=75 --gc=false --uuid="${UUID}" --alerting=false
+  oc delete ns -l kube-burner-uuid="${UUID}"
+  trap - ERR
+}
+
+@test "Running cluster-density timeout case with indexing" {
+  run timeout 600 kube-burner ocp cluster-density --iterations=1 --churn-duration=5m "${COMMON_FLAGS}" --timeout=1s
```
Comment (rsevilla87) [inline test/test-ocp.bats:89]: I think we can remove this timeout

File: mkdocs.yml [hunk 2]
```diff
@@ -66,6 +74,11 @@ markdown_extensions:
   - md_in_html
   - toc:
       permalink: true
+  - pymdownx.superfences:
+      custom_fences:
+        - name: mermaid
+          class: mermaid
+          format: !!python/name:pymdownx.superfences.fence_code_format
```
Comment (rsevilla87) [inline mkdocs.yml:81]: dupe, it's already declared below

Comment (morenod) [reply mkdocs.yml]: deleted

File: .github/workflows/builders.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    - name: Set up Go 1.19
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha }}
```
Comment (rsevilla87) [inline .github/workflows/builders.yml:18]:
```suggestion
    - uses: actions/checkout@v3
      with:
        fetch-depth: 0
        ref: ${{ github.event.pull_request.head.sha }}
        persist-credentials: false
```

According to docs: The auth token is persisted in the local git config. This enables your scripts to run authenticated git commands. The token is removed during post-job cleanup. Set persist-credentials: false to opt-out.

File: .github/workflows/test-k8s.yml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        name: kube-burner
+        path: /tmp/
+
+    - name: Install bats
+      uses: mig4/setup-bats@v1
+      with:
+        bats-version: 1.9.0
+
+    - name: Install oc
+      uses: redhat-actions/oc-installer@v1
```
Comment (rsevilla87) [inline .github/workflows/test-k8s.yml:43]: Do we need to install oc in this workflow?

File: test/test-k8s.bats [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  INDEXING=false kube-burner init -c kube-burner.yml --uuid="${UUID}" --log-level=debug
+}
+
+@test "Running kube-burner: indexing on. pod latency metrics" {
+  INDEXING=true LATENCY_ONLY=true kube-burner init -c kube-burner.yml --uuid="${UUID}" --log-level=debug
+  run test_init_checks
+  [ "$status" -eq 0 ]
+}
+
+@test "Running kube-burner init: indexing on. pod latency mtrics and alert" {
```
Comment (rsevilla87) [inline test/test-k8s.bats:40]:
```suggestion
@test "Running kube-burner init: indexing, pod latency metrics and alerting" {
```

File: test/test-k8s.bats [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  INDEXING=true kube-burner init -c kube-burner.yml --uuid="${UUID}" --log-level=debug -u http://localhost:9090 -m metrics-profile.yaml -a alert-profile.yaml
+  run test_init_checks
+  [ "$status" -eq 0 ]
+}
+
+@test "Running kube-burner init: indexing on. multiple endopoints" {
```
Comment (rsevilla87) [inline test/test-k8s.bats:46]:
```suggestion
@test "Running kube-burner init: indexing and metrics-endpoint" {
```

File: test/test-k8s.bats [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  kubectl delete ns -l kube-burner-uuid="${UUID}" --ignore-not-found
+}
+
+teardown_file() {
+  load helpers.bash
+  destroy-kind
+  podman rm -f prometheus
+}
+
+@test "Running kube-burner: no indexing" {
```
Comment (rsevilla87) [inline test/test-k8s.bats:30]: Can we remove the "Running" prefix from the different testcases?

File: test/test-k8s.bats [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  [ "$status" -eq 0 ]
+}
+
+@test "Running kube-burner init: indexing on. multiple endopoints" {
+  INDEXING=true kube-burner init -c kube-burner.yml --uuid="${UUID}" --log-level=debug -e metrics-endpoints.yaml
+  run test_init_checks
+  [ "$status" -eq 0 ]
+}
+
+@test "Running kube-burner index: test with single prometheus endpoint" {
```
Comment (rsevilla87) [inline test/test-k8s.bats:52]:
```suggestion
@test "Running kube-burner index: metrics-endpoints with single prometheus endpoint" {
```

File: test/test-ocp.bats [hunk 9]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bats
+# vi: ft=bash
+
+setup_file() {
+  export BATS_TEST_TIMEOUT=600
+  export TEMP_FOLDER; TEMP_FOLDER=$(mktemp -d)
+  export JOB_ITERATIONS=5
+  export QPS=2
+  export BURST=2
```
Comment (rsevilla87) [inline test/test-ocp.bats:9]: Not used

Comment (rsevilla87) [inline test/test-ocp.bats:9]: Not used

File: test/test-ocp.bats [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  run kube-burner ocp node-density-heavy --pods-per-node=75 "${COMMON_FLAGS}" --qps=5 --burst=5
+  run check_metric_value etcdVersion
+  [ "$status" -eq 0 ]
+  run check_metric_value clusterMetadata
+  [ "$status" -eq 0 ]
+  run check_metric_value jobSummary
+  [ "$status" -eq 0 ]
+}
+
+@test "Running cluster-density wrapper and user metadata with indexing" {
```
Comment (rsevilla87) [inline test/test-ocp.bats:46]:
```suggestion
@test "cluster-density with user metadata and indexing" {
```

File: test/test-ocp.bats [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  run check_metric_value etcdVersion
+  [ "$status" -eq 0 ]
+  run check_metric_value clusterMetadata
+  [ "$status" -eq 0 ]
+  run check_metric_value jobSummary
+  [ "$status" -eq 0 ]
+}
+
+@test "Running node-density-cni wrapper with gc=false" {
+  # Disable gc and avoid metric indexing
```
Comment (rsevilla87) [inline test/test-ocp.bats:81]:
```suggestion
@test "node-density-cni wrapper with gc=false and no indexing" {
```

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Added some comments,

File: .github/workflows/builders.yml (hunk 3 above)
Comment (morenod) [reply .github/workflows/builders.yml]: fixed

File: .github/workflows/test-k8s.yml (hunk 4 above)
Comment (morenod) [reply .github/workflows/test-k8s.yml]: it was required by an oc command on the tear down, so moved that command to kubectl and removed the step of installing oc

File: test/test-k8s.bats (hunk 5 above)
Comment (morenod) [reply test/test-k8s.bats:40]: done

File: test/test-k8s.bats (hunk 6 above)
Comment (morenod) [reply test/test-k8s.bats:46]: done

File: test/test-k8s.bats (hunk 7 above)
Comment (morenod) [reply test/test-k8s.bats:30]: done

File: test/test-ocp.bats [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+  run check_metric_value etcdVersion
+  [ "$status" -eq 0 ]
+  run check_metric_value clusterMetadata
+  [ "$status" -eq 0 ]
+  run check_metric_value jobSummary
+  [ "$status" -eq 0 ]
+}
+
+@test "Running node-density-heavy wrapper with indexing" {
+  run kube-burner ocp node-density-heavy --pods-per-node=75 "${COMMON_FLAGS}" --qps=5 --burst=5
```
[the remark this answers is not in the document]
Comment (morenod) [reply test/test-ocp.bats:37]: done

File: test/test-ocp.bats (hunk 9 above)
Comment (morenod) [reply test/test-ocp.bats:9]: removed

File: test/test-ocp.bats (hunk 10 above)
Comment (morenod) [reply test/test-ocp.bats:46]: done

File: test/test-ocp.bats (hunk 11 above)
Comment (morenod) [reply test/test-ocp.bats:81]: done

**Review by rsevilla87 — APPROVED** [review]
LGTM, good stuff @morenod !

**Review by vishnuchalla — APPROVED** [review]
Great work. Looks good to me.

**Pull request merged**